File: LeetCode-August-Challenge/MinimumNumberofRefuelingStops.java
import java.util.*;
class Solution {
    public int minRefuelStops(int target, int startFuel, int[][] stations) {
        int maxReach = startFuel, count = 0, index = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        while(maxReach < target){
            while(index != stations.length && stations[index][0] <= maxReach){
                pq.add(stations[index][1]);
                index++;
            }
            if(pq.isEmpty()) return -1;
            maxReach += pq.poll();
            count++;
        }
        return count;
    }
}
public class MinimumNumberofRefuelingStops {
    public static void main(String[] args) {
        int target = 100, startFuel = 1;
        int[][] stations = {{10,100}};
        Solution s = new Solution();
        System.out.println(s.minRefuelStops(target, startFuel, stations));
    }
}

File: Year-2022/LeetCode-August-Challenge/MinimumNumberofRefuelingStops.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int minRefuelStops(int target, int startFuel, vector<vector<int>>& stations) {
        int maxReach = startFuel, count = 0 , index = 0;
        priority_queue<int> pq;
        while(maxReach < target){
            while(index != stations.size() && stations[index][0] <= maxReach){
                pq.push(stations[index][1]);
                index++;
            }
            if(pq.empty()) return -1;
            maxReach += pq.top();
            pq.pop();
            count++;
        }
        return count;
    }
};
int main(){
    int target = 100, startFuel = 1;
    vector<vector<int>> stations = {{10,100}};
    Solution s;
    cout<<s.minRefuelStops(target, startFuel, stations)<<endl;
    return 0;
}